envpool/sokoban/sokoban_node.cc

```diff
@@ -157,12 +157,26 @@ float SokobanNode::GoalDistanceEstimate(SokobanNode& goal_node) {
   float h = 0;
   for (const auto& box : boxes) {
     float min_distance = std::numeric_limits<float>::max();
+    bool goal_along_x = false;
+    bool goal_along_y = false;
     for (const auto& goal_box : goal_node.boxes) {
       float distance =
           abs(box.first - goal_box.first) + abs(box.second - goal_box.second);
       min_distance = std::min(min_distance, distance);
+      if (box.first == goal_box.first) {
+        goal_along_y = true;
+      }
+      if (box.second == goal_box.second) {
+        goal_along_x = true;
+      }
     }
     h += min_distance;
+    int surr_walls = SurroundingWalls(box);
+    if (surr_walls > 1 && min_distance != 0) {
+      h += 10;
```

Comment:
AFAICT this is not an [admissible heuristic](https://en.wikipedia.org/wiki/Admissible_heuristic). Why does the cost increase by 10 when there's 1 wall next to the box? This needs an explanation.

Reply:
If you mean to say " impossible" you should probably add a lot more than 10, like 1000.